### PR TITLE
[#5] Implement pretty Show instance for Memory

### LINKED ISSUE #5: Pretty Show instance

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #5 

I haven't added documentation to the existing functions (as we have separate `doctest` issue).

Because of GHC incompleteness, current instances of `UnitSymbol` doesn't look very pretty... But not much we can do here. We can write type-level tests later.

### CODE REVIEW

File: src/Membrain/Units.hs [hunk 1]
```diff
@@ -89,97 +73,35 @@ type Exbibyte  = 1024 * Pebibyte
...
+type instance UnitSymbol 8000000000000000000000000 = "YB"
+
+type instance UnitSymbol 8192                      = "KiB"
+type instance UnitSymbol 8388608                   = "MiB"
+type instance UnitSymbol 8589934592                = "GiB"
+type instance UnitSymbol 8796093022208             = "TiB"
+type instance UnitSymbol 9007199254740992          = "PiB"
+type instance UnitSymbol 9223372036854775808       = "EiB"
+type instance UnitSymbol 9444732965739290427392    = "ZiB"
+type instance UnitSymbol 9671406556917033397649408 = "YiB"
```
Comment (vrom911) [inline src/Membrain/Units.hs:99]: 😿

**Review by vrom911 — APPROVED** [review]
Wow! Great job! 👍

File: src/Membrain/Memory.hs [hunk 2]
```diff
@@ -26,13 +28,49 @@ import Data.Ratio (Ratio, (%))
...
+{- | 'Show' instance for 'Memory' shows value as 'Double' with measure unit
+suffix. It shows 'Memory' losslessly while used with standardized units of
+measures. The following math fact is used to display 'Memory'.
+
+A decimal representation written with a repeating final @0@ is said to terminate
+before these zeros. Instead of @1.585000...@ one simply writes @1.585@. The
+decimal is also called a terminating decimal. Terminating decimals represent
+rational numbers of the form  \( \cfrac{k}{2^n 5^m} \).
+
+If you use unconventional units, this operation will hang. TODO: check this statically
```
Comment (vrom911) [inline src/Membrain/Memory.hs]: The word `unconventional` here is a bit tricky, is it unconventional for our library, or generally?

Comment (chshersh) [reply src/Membrain/Memory.hs]: I will clarify that!

**Review by vrom911 — APPROVED** [review]

**Pull request merged**